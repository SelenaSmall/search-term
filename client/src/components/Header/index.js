import React from 'react';

export default ({ status, score, secondsRemaining}) => (
  <section id="top-region">
    <menu className="status-menu">
      <ul>
        <li className="status-menu-item status">
          <span className="status-menu-item-label">Status</span>
          <span className="status-menu-item-value">{status}</span>
        </li>
        <li className="status-menu-item score">
          <span className="status-menu-item-label">Score</span>
          <span className="status-menu-item-value">{score}</span>
        </li>
        <li className="status-menu-item timer">
          <span className="status-menu-item-label">Time</span>
          <span className="status-menu-item-value">{secondsRemaining}</span>
        </li>
      </ul>
    </menu>
  </section>
);
