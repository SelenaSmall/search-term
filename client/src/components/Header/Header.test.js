import React from 'react';
import Header from './';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

it('basic rendering', () => {
  const output = shallow(<Header />)
  expect(shallowToJson(output)).toMatchSnapshot()
});

it('renders status, score and time remaining', () => {
  const output = shallow(<Header status={'the status'}/>)
  expect(shallowToJson(output)).toMatchSnapshot()
  const outputWithScore = shallow(<Header status={'the status'} score={101} />)
  expect(shallowToJson(outputWithScore)).toMatchSnapshot()
  const outputWithSecondsRemaining = shallow(<Header status={'the status'} score={101} secondsRemaining={56} />)
  expect(shallowToJson(outputWithSecondsRemaining)).toMatchSnapshot()
});
