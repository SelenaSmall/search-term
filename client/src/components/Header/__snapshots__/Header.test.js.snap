// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering 1`] = `
<section
  id="top-region"
>
  <menu
    className="status-menu"
  >
    <ul>
      <li
        className="status-menu-item status"
      >
        <span
          className="status-menu-item-label"
        >
          Status
        </span>
        <span
          className="status-menu-item-value"
        />
      </li>
      <li
        className="status-menu-item score"
      >
        <span
          className="status-menu-item-label"
        >
          Score
        </span>
        <span
          className="status-menu-item-value"
        />
      </li>
      <li
        className="status-menu-item timer"
      >
        <span
          className="status-menu-item-label"
        >
          Time
        </span>
        <span
          className="status-menu-item-value"
        />
      </li>
    </ul>
  </menu>
</section>
`;

exports[`renders status, score and time remaining 1`] = `
<section
  id="top-region"
>
  <menu
    className="status-menu"
  >
    <ul>
      <li
        className="status-menu-item status"
      >
        <span
          className="status-menu-item-label"
        >
          Status
        </span>
        <span
          className="status-menu-item-value"
        >
          the status
        </span>
      </li>
      <li
        className="status-menu-item score"
      >
        <span
          className="status-menu-item-label"
        >
          Score
        </span>
        <span
          className="status-menu-item-value"
        />
      </li>
      <li
        className="status-menu-item timer"
      >
        <span
          className="status-menu-item-label"
        >
          Time
        </span>
        <span
          className="status-menu-item-value"
        />
      </li>
    </ul>
  </menu>
</section>
`;

exports[`renders status, score and time remaining 2`] = `
<section
  id="top-region"
>
  <menu
    className="status-menu"
  >
    <ul>
      <li
        className="status-menu-item status"
      >
        <span
          className="status-menu-item-label"
        >
          Status
        </span>
        <span
          className="status-menu-item-value"
        >
          the status
        </span>
      </li>
      <li
        className="status-menu-item score"
      >
        <span
          className="status-menu-item-label"
        >
          Score
        </span>
        <span
          className="status-menu-item-value"
        >
          101
        </span>
      </li>
      <li
        className="status-menu-item timer"
      >
        <span
          className="status-menu-item-label"
        >
          Time
        </span>
        <span
          className="status-menu-item-value"
        />
      </li>
    </ul>
  </menu>
</section>
`;

exports[`renders status, score and time remaining 3`] = `
<section
  id="top-region"
>
  <menu
    className="status-menu"
  >
    <ul>
      <li
        className="status-menu-item status"
      >
        <span
          className="status-menu-item-label"
        >
          Status
        </span>
        <span
          className="status-menu-item-value"
        >
          the status
        </span>
      </li>
      <li
        className="status-menu-item score"
      >
        <span
          className="status-menu-item-label"
        >
          Score
        </span>
        <span
          className="status-menu-item-value"
        >
          101
        </span>
      </li>
      <li
        className="status-menu-item timer"
      >
        <span
          className="status-menu-item-label"
        >
          Time
        </span>
        <span
          className="status-menu-item-value"
        >
          56
        </span>
      </li>
    </ul>
  </menu>
</section>
`;
